compiler/shared_ast/scope.mli

```diff
@@ -44,6 +44,7 @@ val fold_right_lets :
     scope lets to be examined (which are before in the program order). *)
 
 val map_exprs_in_lets :
+  ?reset_types:bool ->
```

Comment:
An explanation of this arg maybe ? Esp. since it clears the `ScopeLet` types, but doesn't untype the expressions.